src/app/modules/group/components/group-invite-users/group-invite-users.component.ts

```diff
@@ -81,27 +101,14 @@ export class GroupInviteUsersComponent {
   }
 
   /* events */
-
-  onTextChange(text:string): void {
-    this.state = 'ready';
-
-    const logins = text.split(',').filter(login => login.length > 0);
-
-    if (logins.length === 0) {
-      this.state = 'empty';
-    } else if (logins.length >= 100) {
-      this.state = 'too_many';
-    }
-  }
-
   onInviteClicked(): void {
-    if (!this.textArea || !this.group) return;
+    if (!this.group) return;
```

Comment:
Done

Should this kind of failsafe be implemented every times? The button calling this function is only enabled when state == ready, so a regular user isn't supposed to find itself calling this function if the state isn't ready.

Reply:
Several reasons:
- Here, it is not completely a responsibility of this class. It is a public method, an external class (or the template) may call this function at a wrong moment. 
- Sometimes, the fact a function is called even we are not supposed to is not a problem as the action is not harmful.
- By the important software engineering principle of "fail-fast", it is important to catch unexpected behavior as soon as possible. Otherwise we may get behavior very complicated to debug if this happens.
A nice class design tries to make unconsistent cases impossible to happen (e.g., it would be impossible to call action in case the state is loading). That is not always possible to achieve this, if so, we have to add guards.

Reply:
Ok, thanks !